### PR TITLE
Adjust the structure of next/prev URLs so they can be properly tested

### PR DESCRIPTION
**Pull request by borkweb**
The structure of the next/prev URLs in View were such that tests around the filterability of the query arguments would fail. The slight restructure moves the query arguments to an array that gets filtered rather than adding/removing from the url over and over.

Related: https://github.com/moderntribe/events-filterbar/pull/228

:movie_camera: http://p.tri.be/cXR0UB

:ticket: [FBAR-64] (sorta related)

[FBAR-64]: https://moderntribe.atlassian.net/browse/FBAR-64

### CODE REVIEW

File: src/Tribe/Views/V2/View.php [hunk 1]
```diff
@@ -755,25 +755,37 @@ public function next_url( $canonical = false, array $passthru_vars = [] ) {
 
 		$url = $this->get_url();
 
+		$query_args = [];
```
Comment (tr1b0t) [inline src/Tribe/Views/V2/View.php]: :no_entry_sign: Short array syntax is not allowed

<sub>hash: 134e75a5772df713a13580b1deca1580</sub>

File: src/Tribe/Views/V2/View.php [hunk 2]
```diff
@@ -755,25 +755,37 @@ public function next_url( $canonical = false, array $passthru_vars = [] ) {
...
-		$url = add_query_arg( [ 'eventDisplay' => $this->slug ], $url );
+		$query_args['eventDisplay'] = $this->slug;
 
-		$url = $this->has_next_event ?
-			add_query_arg( [ $this->page_key => $this->url->get_current_page() + 1 ], $url )
-			: '';
+		if ( $this->has_next_event ) {
+			$query_args[ $this->page_key ] = $this->url->get_current_page() + 1;
+		} else {
+			$query_args = [];
```
Comment (tr1b0t) [inline src/Tribe/Views/V2/View.php]: :no_entry_sign: Short array syntax is not allowed

<sub>hash: 94dddfa7c470cebbd259353dfcefebce</sub>

File: src/Tribe/Views/V2/View.php [hunk 3]
```diff
@@ -793,8 +805,9 @@ public function prev_url( $canonical = false, array $passthru_vars = [] ) {
 
 		$prev_page  = $this->repository->prev()->order_by( '__none' );
 
-		$paged      = $this->url->get_current_page() - 1;
-		$query_args = $paged > 1
+		$paged           = $this->url->get_current_page() - 1;
+		$query_args      = [];
```
Comment (tr1b0t) [inline src/Tribe/Views/V2/View.php]: :no_entry_sign: Short array syntax is not allowed

<sub>hash: a67d07b7a9a050b0f83261e27a5aac60</sub>

File: src/Tribe/Views/V2/View.php [hunk 4]
```diff
@@ -806,12 +819,18 @@ public function prev_url( $canonical = false, array $passthru_vars = [] ) {
...
 
 		// Make sure the view slug is always set to correctly match rewrites.
-		$url = add_query_arg( [ 'eventDisplay' => $this->slug ], $url );
+		$query_args['eventDisplay'] = $this->slug;
 
-		$url = $prev_page->count() > 0 ? add_query_arg( $query_args, $url ) : '';
+		if ( $prev_page->count() > 0 ) {
+			$query_args = array_merge( $query_args, $page_query_args );
+		} else {
+			$query_args = [];
```
Comment (tr1b0t) [inline src/Tribe/Views/V2/View.php]: :no_entry_sign: Short array syntax is not allowed

<sub>hash: 3a553ed58ef2331c115416000c239246</sub>

File: src/Tribe/Views/V2/View.php (hunk 1 above)
Comment (tr1b0t) [inline src/Tribe/Views/V2/View.php]: :no_entry_sign: Spaces must be used to indent lines; tabs are not allowed

<sub>hash: 441593dbe34e42e31b9d91393ccea923</sub>

File: src/Tribe/Views/V2/View.php [hunk 5]
```diff
@@ -755,25 +755,37 @@ public function next_url( $canonical = false, array $passthru_vars = [] ) {
...
+		$query_args = [];
+
 		if ( ! empty( $passthru_vars ) ) {
 			// Remove the pass-thru vars, we'll re-apply them to the URL later.
 			$url = remove_query_arg( array_keys( $passthru_vars ), $url );
 		}
 
 		// Make sure the view slug is always set to correctly match rewrites.
-		$url = add_query_arg( [ 'eventDisplay' => $this->slug ], $url );
+		$query_args['eventDisplay'] = $this->slug;
```
Comment (tr1b0t) [inline src/Tribe/Views/V2/View.php]: :no_entry_sign: Spaces must be used to indent lines; tabs are not allowed

<sub>hash: f87c405dcbb7a2c826b9bf383d16f8ea</sub>

File: src/Tribe/Views/V2/View.php [hunk 6]
```diff
@@ -755,25 +755,37 @@ public function next_url( $canonical = false, array $passthru_vars = [] ) {
...
 
-		$url = $this->has_next_event ?
-			add_query_arg( [ $this->page_key => $this->url->get_current_page() + 1 ], $url )
-			: '';
+		if ( $this->has_next_event ) {
```
Comment (tr1b0t) [inline src/Tribe/Views/V2/View.php]: :no_entry_sign: Spaces must be used to indent lines; tabs are not allowed

<sub>hash: cbd176cff4adc40dd974a9ea5c394b85</sub>

File: src/Tribe/Views/V2/View.php [hunk 7]
```diff
@@ -755,25 +755,37 @@ public function next_url( $canonical = false, array $passthru_vars = [] ) {
...
+			$query_args[ $this->page_key ] = $this->url->get_current_page() + 1;
```
Comment (tr1b0t) [inline src/Tribe/Views/V2/View.php]: :no_entry_sign: Spaces must be used to indent lines; tabs are not allowed

<sub>hash: 221eb070cf2925663303d93d895fbba6</sub>

File: src/Tribe/Views/V2/View.php [hunk 8]
```diff
@@ -755,25 +755,37 @@ public function next_url( $canonical = false, array $passthru_vars = [] ) {
...
+		} else {
```
Comment (tr1b0t) [inline src/Tribe/Views/V2/View.php]: :no_entry_sign: Spaces must be used to indent lines; tabs are not allowed

<sub>hash: dc48c9cd191a0ca0803c16d53289c496</sub>

File: src/Tribe/Views/V2/View.php (hunk 2 above)
Comment (tr1b0t) [inline src/Tribe/Views/V2/View.php]: :no_entry_sign: Spaces must be used to indent lines; tabs are not allowed

<sub>hash: f91eb646d66021433c8be82dc7d8ad3a</sub>

File: src/Tribe/Views/V2/View.php [hunk 9]
```diff
@@ -755,25 +755,37 @@ public function next_url( $canonical = false, array $passthru_vars = [] ) {
...
+			$query_args = [];
+			$url        = '';
```
Comment (tr1b0t) [inline src/Tribe/Views/V2/View.php]: :no_entry_sign: Spaces must be used to indent lines; tabs are not allowed

<sub>hash: e5b0191fc8aac42fcd7a3ea5d02322ec</sub>

File: src/Tribe/Views/V2/View.php [hunk 10]
```diff
@@ -755,25 +755,37 @@ public function next_url( $canonical = false, array $passthru_vars = [] ) {
...
+		}
```
Comment (tr1b0t) [inline src/Tribe/Views/V2/View.php]: :no_entry_sign: Spaces must be used to indent lines; tabs are not allowed

<sub>hash: 67679ef4ce8eee3032179add10fc9f63</sub>

File: src/Tribe/Views/V2/View.php [hunk 11]
```diff
@@ -755,25 +755,37 @@ public function next_url( $canonical = false, array $passthru_vars = [] ) {
...
 
 		if ( ! empty( $url ) && $canonical ) {
 			$url = tribe( 'events.rewrite' )->get_clean_url( $url );
 		}
 
 		if ( ! empty( $passthru_vars ) && ! empty( $url ) ) {
 			// Re-apply the pass-thru query arguments.
-			$url = add_query_arg( $passthru_vars, $url );
+			$query_args = array_merge( $query_args, $passthru_vars );
```
Comment (tr1b0t) [inline src/Tribe/Views/V2/View.php]: :no_entry_sign: Spaces must be used to indent lines; tabs are not allowed

<sub>hash: 2c55ddd00ce7aee0b9b9ba351d57cf1d</sub>

File: src/Tribe/Views/V2/View.php [hunk 12]
```diff
@@ -755,25 +755,37 @@ public function next_url( $canonical = false, array $passthru_vars = [] ) {
...
+		}
```
Comment (tr1b0t) [inline src/Tribe/Views/V2/View.php]: :no_entry_sign: Spaces must be used to indent lines; tabs are not allowed

<sub>hash: 97c7dfa199adf1c08251a8e5c3a932c0</sub>

File: src/Tribe/Views/V2/View.php [hunk 13]
```diff
@@ -755,25 +755,37 @@ public function next_url( $canonical = false, array $passthru_vars = [] ) {
...
+
+		$query_args = $this->filter_query_args( $query_args, $url );
```
Comment (tr1b0t) [inline src/Tribe/Views/V2/View.php]: :no_entry_sign: Spaces must be used to indent lines; tabs are not allowed

<sub>hash: 4688b27fc965067cfeafbfb8174e95f1</sub>

File: src/Tribe/Views/V2/View.php [hunk 14]
```diff
@@ -755,25 +755,37 @@ public function next_url( $canonical = false, array $passthru_vars = [] ) {
...
+		$query_args = array_filter( $query_args );
```
Comment (tr1b0t) [inline src/Tribe/Views/V2/View.php]: :no_entry_sign: Spaces must be used to indent lines; tabs are not allowed

<sub>hash: 015c5674921f35921b39e9c0b7adfc61</sub>

File: src/Tribe/Views/V2/View.php [hunk 15]
```diff
@@ -755,25 +755,37 @@ public function next_url( $canonical = false, array $passthru_vars = [] ) {
...
+
+		if ( ! empty( $url ) || ! empty( $query_args ) ) {
```
Comment (tr1b0t) [inline src/Tribe/Views/V2/View.php]: :no_entry_sign: Spaces must be used to indent lines; tabs are not allowed

<sub>hash: dc2a3f701b86c8f145c96a70def22b08</sub>

File: src/Tribe/Views/V2/View.php [hunk 16]
```diff
@@ -755,25 +755,37 @@ public function next_url( $canonical = false, array $passthru_vars = [] ) {
...
+			$url = add_query_arg( $query_args, $url );
```
Comment (tr1b0t) [inline src/Tribe/Views/V2/View.php]: :no_entry_sign: Spaces must be used to indent lines; tabs are not allowed

<sub>hash: f47045edc2a3259714bb50e8a4dbb210</sub>

File: src/Tribe/Views/V2/View.php [hunk 17]
```diff
@@ -793,8 +805,9 @@ public function prev_url( $canonical = false, array $passthru_vars = [] ) {
 
 		$prev_page  = $this->repository->prev()->order_by( '__none' );
 
-		$paged      = $this->url->get_current_page() - 1;
-		$query_args = $paged > 1
+		$paged           = $this->url->get_current_page() - 1;
```
Comment (tr1b0t) [inline src/Tribe/Views/V2/View.php]: :no_entry_sign: Spaces must be used to indent lines; tabs are not allowed

<sub>hash: c760b481f966b1acaa9f70824a2ac91c</sub>

File: src/Tribe/Views/V2/View.php (hunk 3 above)
Comment (tr1b0t) [inline src/Tribe/Views/V2/View.php]: :no_entry_sign: Spaces must be used to indent lines; tabs are not allowed

<sub>hash: 393d1683914e0d780296d8643b6adbe6</sub>

File: src/Tribe/Views/V2/View.php [hunk 18]
```diff
@@ -793,8 +805,9 @@ public function prev_url( $canonical = false, array $passthru_vars = [] ) {
...
+		$query_args      = [];
+		$page_query_args = $paged > 1
```
Comment (tr1b0t) [inline src/Tribe/Views/V2/View.php]: :no_entry_sign: Spaces must be used to indent lines; tabs are not allowed

<sub>hash: cfaefc5726c47c331c2c884474b97a91</sub>

File: src/Tribe/Views/V2/View.php [hunk 19]
```diff
@@ -806,12 +819,18 @@ public function prev_url( $canonical = false, array $passthru_vars = [] ) {
 		}
 
 		// Make sure the view slug is always set to correctly match rewrites.
-		$url = add_query_arg( [ 'eventDisplay' => $this->slug ], $url );
+		$query_args['eventDisplay'] = $this->slug;
```
Comment (tr1b0t) [inline src/Tribe/Views/V2/View.php]: :no_entry_sign: Spaces must be used to indent lines; tabs are not allowed

<sub>hash: f35a3c4b89eca3aafc46c15dd13ca746</sub>

File: src/Tribe/Views/V2/View.php [hunk 20]
```diff
@@ -806,12 +819,18 @@ public function prev_url( $canonical = false, array $passthru_vars = [] ) {
...
 
-		$url = $prev_page->count() > 0 ? add_query_arg( $query_args, $url ) : '';
+		if ( $prev_page->count() > 0 ) {
```
Comment (tr1b0t) [inline src/Tribe/Views/V2/View.php]: :no_entry_sign: Spaces must be used to indent lines; tabs are not allowed

<sub>hash: 96fd42b34267d0380436f3640d578e65</sub>

File: src/Tribe/Views/V2/View.php [hunk 21]
```diff
@@ -806,12 +819,18 @@ public function prev_url( $canonical = false, array $passthru_vars = [] ) {
...
+			$query_args = array_merge( $query_args, $page_query_args );
```
Comment (tr1b0t) [inline src/Tribe/Views/V2/View.php]: :no_entry_sign: Spaces must be used to indent lines; tabs are not allowed

<sub>hash: 57401497548bacb6dde799eabb7b24b9</sub>

File: src/Tribe/Views/V2/View.php [hunk 22]
```diff
@@ -806,12 +819,18 @@ public function prev_url( $canonical = false, array $passthru_vars = [] ) {
...
+		} else {
```
Comment (tr1b0t) [inline src/Tribe/Views/V2/View.php]: :no_entry_sign: Spaces must be used to indent lines; tabs are not allowed

<sub>hash: 03f47ed42de689656c3d76886f1eec21</sub>

File: src/Tribe/Views/V2/View.php (hunk 4 above)
Comment (tr1b0t) [inline src/Tribe/Views/V2/View.php]: :no_entry_sign: Spaces must be used to indent lines; tabs are not allowed

<sub>hash: 99daad3e433f9f9ccef90e8b1b675548</sub>

File: src/Tribe/Views/V2/View.php [hunk 23]
```diff
@@ -806,12 +819,18 @@ public function prev_url( $canonical = false, array $passthru_vars = [] ) {
...
+			$query_args = [];
+			$url        = '';
```
Comment (tr1b0t) [inline src/Tribe/Views/V2/View.php]: :no_entry_sign: Spaces must be used to indent lines; tabs are not allowed

<sub>hash: c62f7696340adb4b072655a3aae3bceb</sub>

File: src/Tribe/Views/V2/View.php [hunk 24]
```diff
@@ -806,12 +819,18 @@ public function prev_url( $canonical = false, array $passthru_vars = [] ) {
...
+		}
```
Comment (tr1b0t) [inline src/Tribe/Views/V2/View.php]: :no_entry_sign: Spaces must be used to indent lines; tabs are not allowed

<sub>hash: 777ea4eb70a066cbd31ca25b88c6d35b</sub>

File: src/Tribe/Views/V2/View.php [hunk 25]
```diff
@@ -806,12 +819,18 @@ public function prev_url( $canonical = false, array $passthru_vars = [] ) {
...
 
 		if ( ! empty( $url ) && $paged === 1 ) {
 			$url = remove_query_arg( $this->page_key, $url );
+			unset( $query_args[ $this->page_key ] );
```
Comment (tr1b0t) [inline src/Tribe/Views/V2/View.php]: :no_entry_sign: Spaces must be used to indent lines; tabs are not allowed

<sub>hash: 6eedb1df61eda7b4cbe043e68ff8c060</sub>

File: src/Tribe/Views/V2/View.php [hunk 26]
```diff
@@ -820,7 +839,14 @@ public function prev_url( $canonical = false, array $passthru_vars = [] ) {
 
 		if ( ! empty( $passthru_vars ) && ! empty( $url ) ) {
 			// Re-apply the pass-thru query arguments.
-			$url = add_query_arg( $passthru_vars, $url );
+			$query_args = array_merge( $query_args, $passthru_vars );
```
Comment (tr1b0t) [inline src/Tribe/Views/V2/View.php]: :no_entry_sign: Spaces must be used to indent lines; tabs are not allowed

<sub>hash: 5f32cdd1e7eeba7f4d9dbb2af77c366a</sub>

File: src/Tribe/Views/V2/View.php [hunk 27]
```diff
@@ -820,7 +839,14 @@ public function prev_url( $canonical = false, array $passthru_vars = [] ) {
...
+		}
```
Comment (tr1b0t) [inline src/Tribe/Views/V2/View.php]: :no_entry_sign: Spaces must be used to indent lines; tabs are not allowed

<sub>hash: 0b73a354826af5ae26860f96cfd70770</sub>

File: src/Tribe/Views/V2/View.php [hunk 28]
```diff
@@ -820,7 +839,14 @@ public function prev_url( $canonical = false, array $passthru_vars = [] ) {
...
+
+		$query_args = $this->filter_query_args( $query_args, $url );
```
Comment (tr1b0t) [inline src/Tribe/Views/V2/View.php]: :no_entry_sign: Spaces must be used to indent lines; tabs are not allowed

<sub>hash: e48ab28d3b29b99140fd9a767badc52f</sub>

File: src/Tribe/Views/V2/View.php [hunk 29]
```diff
@@ -820,7 +839,14 @@ public function prev_url( $canonical = false, array $passthru_vars = [] ) {
...
+		$query_args = array_filter( $query_args );
```
Comment (tr1b0t) [inline src/Tribe/Views/V2/View.php]: :no_entry_sign: Spaces must be used to indent lines; tabs are not allowed

<sub>hash: ec4a31c8a2f5c2923e12b0365c2b2fd2</sub>

File: src/Tribe/Views/V2/View.php [hunk 30]
```diff
@@ -820,7 +839,14 @@ public function prev_url( $canonical = false, array $passthru_vars = [] ) {
...
+
+		if ( ! empty( $url ) || ! empty( $query_args ) ) {
```
Comment (tr1b0t) [inline src/Tribe/Views/V2/View.php]: :no_entry_sign: Spaces must be used to indent lines; tabs are not allowed

<sub>hash: 3b3dee502ea64075e83a817b7bf49817</sub>

File: src/Tribe/Views/V2/View.php [hunk 31]
```diff
@@ -820,7 +839,14 @@ public function prev_url( $canonical = false, array $passthru_vars = [] ) {
...
+			$url = add_query_arg( $query_args, $url );
```
Comment (tr1b0t) [inline src/Tribe/Views/V2/View.php]: :no_entry_sign: Spaces must be used to indent lines; tabs are not allowed

<sub>hash: d00a83257a6cd9b27ed4acefa8f932ef</sub>

**Comment by borkweb** [comment]
Tests that are failing are those that fail due to timezone stuff after 5pm ET. Merging.

**Pull request merged**